Order Transaction Types by 'fixed' and then 'text' (3059)

sfount opened the pull request:
This commit orders the Transaction Type Management `ui-grid`. It first
orders by the 'fixed' property (if the transaction type is part of the
system of a custom user type) and then alphabetically.

It also updates the `ui-grid` according to a number of other similar UI patterns found in the application.

Addresses a user facing bug resulting in the Transaction Type 'types' rendering hard-coded database values. 

**Current master:**
![screenshot from 2018-08-30 17-47-08](https://user-images.githubusercontent.com/2844572/44866439-28312900-ac7d-11e8-99fd-dba9c1ef7a18.png)

Note: French translation with English hard-coded values 

**Updated branch:**
![screenshot from 2018-08-30 17-35-57](https://user-images.githubusercontent.com/2844572/44866467-3aab6280-ac7d-11e8-8d1e-fd5d868f65ca.png)

Closes https://github.com/IMA-WorldHealth/bhima-2.X/issues/3059

sfount committed:
chore(transaction-types): order trans. type list

This commit orders the Transaction Type Management `ui-grid`. It first
orders by the 'fixed' property (if the transaction type is part of the
system of a custom user type) and then alphabetically.

sfount committed:
chore(transaction-types): make cell template file

This commit moves the `ui-grid` cell template into it's own file
allowing the HTML to be edited properly. It also:
* Make 'fixed' transaction type actions a `span` instead of `a`. They
cannot be clicked and this is a clearer UI-pattern.
* Move translation out of `$translate.instant` - this uses the wrong
language if the page hasn't fully loaded by the time it's called
* Update style of Actions column to fit other `ui-grid` styles
* Minor lint improvements

sfount committed:
fix(transaction-types): fix type translation issue

This commit fixes a bug on the Transaction Type label in Transaction
Type Management.

Previously the hardcoded `type` value from the database was being
displayed - this was not well formatted and is exclusively in English.
This fix formally defines a constant transaction type map in
`bhConstants` and shares this among controllers that need to dynamically
show types.

jniles commented on client/src/js/constants/bhConstants.js:
```diff
@@ -134,6 +144,10 @@ function constantConfig() {
       { id : 6, label : 'FORM.LABELS.WEEK_DAYS.SATURDAY' },
     ],
+    transactionTypesMap : {
+      income : { label : 'VOUCHERS.SIMPLE.INCOME', value : 'income' },
```
It probably is better English to call this "transactionTypeMap".

jniles commented on client/src/modules/transaction-type/transaction-type.ctrl.js:
```diff
@@ -96,25 +84,32 @@ function TransactionTypeController(TransactionType, TransactionTypeStore, Notify
+    // @TODO(sfount) The `transaction_type` database currently hard codes 'income',
+    //               'expense' and 'other' transaction types. When this is updated
+    //               with a more data driven approach it should include translatable
+    //               labels
```
Good comment!  Mind making an issue about it?  I can't recall if one exists or not..

jniles reviewed (commented):
@sfount great work!  I hadn't realized that the transaction types weren't respecting i18n.

Could we get one more level of sort on here?  It would be nice to sort the left-most values (`field: text`) in alphabetical order.   You'll probably need to translate the values programmatically and then sort them.  That way, it is easy to locate a particular transaction type in the grid.

sfount committed:
fix(transaction-types): fix type description sort

This commit fixes an issue with sorting transaction type descriptions.
The 'Text' column was previously being sorted on the translation key,
moving the translation into the controller allows UI grid to sort on the
display value instead of the key.

jniles reviewed (approved):
Looks good to me!  Great work.

bors r+
